### PR TITLE
[Token] auction house add token withdraw with proof  

### PR DESCRIPTION
**Pull request by areshand**
### Description

add withdraw with capability
### Test Plan
tested in https://github.com/aptos-labs/aptos-core/pull/3276

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4262)

### CODE REVIEW

File: aptos-move/framework/aptos-token/sources/token.move [hunk 1]
```diff
@@ -586,6 +604,36 @@ module aptos_token::token {
         direct_deposit(to, token);
     }
 
+
+    /// Token owner can create this one-time withdraw capability with an expiration time
+    public fun create_withdraw_capability(
```
Comment (movekevin) [inline aptos-move/framework/aptos-token/sources/token.move:609]: Why not extract the token and keep it in the auction's account? Auction house can be hosted in a new resource account.

Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move:609]: no, we want the owner to still have the token until the auction completes. So they can still use/show their token before actual sale happens

Comment (movekevin) [reply aptos-move/framework/aptos-token/sources/token.move:609]: But what if they then transfer it somewhere else? Wouldn't the auction not be honored?

Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move:609]: The auction will fail. That is the auction house's responsibility to verify the listing

Comment (movekevin) [reply aptos-move/framework/aptos-token/sources/token.move:609]: They can't verify the listing if the token owner transfers close to the end of the auction. This seems a bit brittle in my opinion. What's wrong with the token owner giving up temporary custody during the auction?

Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move:609]: sorry, I think this is quite common in NFT marketplace. people don't put their NFT to the auction house escrow. There are many way for the auction house to ensure the high quality of listing. They can just penalize the user with bad behavior by blacklist them or create a reputation score, etc....

Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move:609]: btw: this discussion is irrelevant to the this PR for supporting withdraw with a capability right?

Comment (davidiw) [reply aptos-move/framework/aptos-token/sources/token.move:609]: @movekevin , while I agree with you, these are product aspects. I'm fine with exploring both paths, but I ultimately really dislike the idea of an auction that doesn't contain the item. It seems to violate my notion of web3 :P

**Review by davidiw — APPROVED** [review]

**Review by chloeqjz — APPROVED** [review]

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-token/sources/token.move:609]: I still think it's strange that the auction can happen and buyers commit money into it, and then the seller just refuses to sell at the end by transferring the token away.

**Pull request merged**